packages/contracts-bedrock/test/L2/SuperchainERC20.t.sol

```diff
@@ -0,0 +1,292 @@
+        // Expect the revert reason "ERC20: mint to the zero address"
+        vm.expectRevert("ERC20: mint to the zero address");
+
+        // Call the `mint` function with the zero address
+        vm.prank(BRIDGE);
+        address _to = address(0);
+        superchainERC20.mint(_to, _amount);
+    }
+
+    /// @dev Tests the `mint` succeeds and emits the `Mint` event.
+    function testFuzz_mint_succeeds(address _to, uint256 _amount) public {
+        // Ensure `_to` is not the zero address
+        vm.assume(_to != address(0));
+
+        // Get the total supply and balance of `_to` before the mint to compare later on the assertions
+        uint256 _totalSupplyBefore = superchainERC20.totalSupply();
+        uint256 _toBalanceBefore = superchainERC20.balanceOf(_to);
+
+        // Look for the emit of the `Transfer` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit IERC20.Transfer(address(0), _to, _amount);
+
+        // Look for the emit of the `Mint` event
+        vm.expectEmit(true, true, true, true, address(superchainERC20));
+        emit SuperchainERC20.Mint(_to, _amount);
+
+        // Call the `mint` function with the bridge caller
+        vm.prank(BRIDGE);
+        superchainERC20.mint(_to, _amount);
+
+        // Check the total supply and balance of `_to` after the mint were updated correctly
+        assertEq(superchainERC20.totalSupply(), _totalSupplyBefore + _amount);
+        assertEq(superchainERC20.balanceOf(_to), _toBalanceBefore + _amount);
+    }
+
+    /// @dev Tests the `burn` function reverts when the caller is not the bridge.
+    function testFuzz_burn_callerNotBridge_reverts(address _caller, address _from, uint256 _amount) public {
+        // Ensure the caller is not the bridge
+        vm.assume(_caller != BRIDGE);
+
+        // Expect the revert with `CallerNotBridge` selector
+        vm.expectRevert(abi.encodeWithSelector(CallerNotBridge.selector));
+
+        // Call the `burn` function with the non-bridge caller
+        vm.prank(_caller);
+        superchainERC20.burn(_from, _amount);
+    }
+
+    /// @dev Tests the `burn` function reverts when the amount is zero.
+    function testFuzz_burn_zeroAddressFrom_reverts(uint256 _amount) public {
+        // Expect the revert reason "ERC20: burn from the zero address"
+        vm.expectRevert("ERC20: burn from the zero address");
+
+        // Call the `burn` function with the zero address
+        vm.prank(BRIDGE);
+        address _from = address(0);
+        superchainERC20.burn(_from, _amount);
+    }
+
+    /// @dev Tests the `burn` succeeds and emits the `Burn` event.
+    function testFuzz_burn_succeeds(address _from, uint256 _amount) public {
+        // Ensure `_from` is not the zero address
+        vm.assume(_from != address(0));
+
+        // Mint some tokens to `_from` so then they can be burned
+        vm.prank(BRIDGE);
+        superchainERC20.mint(_from, _amount);
```

Comment:
`deal()` didn't work here so I used the `mint()`

Reply:
I believe it's okay. I'm not sure if `deal` increases the `totalSupply`, and the `SuperchainERC20` tokens start with a `totalSupply` of `0`. If you don't mint, the `totalSupply` would underflow. Either way, how did you attempt to use `deal` here, and with what error did it fail?

`deal(address(SuperchainERC20), _from, _amount)` didn't work?